UnityProject/Assets/Scripts/Core/Editor/SceneAutoLoader.cs

```diff
@@ -1,100 +1,111 @@
+	/// </description>
+	[InitializeOnLoad]
+	static class SceneAutoLoader
 	{
-		if (!EditorApplication.isPlaying && EditorApplication.isPlayingOrWillChangePlaymode)
+		private static string MasterScene => "Assets/Scenes/ActiveScenes/OnlineScene.unity";
+
+		private static string PreviousSceneName
+		{
+			get => EditorPrefs.GetString("prevEditorScene", UnityEngine.SceneManagement.SceneManager.GetActiveScene().name);
+			set => EditorPrefs.SetString("prevEditorScene", value);
+		}
+
+		private static string PreviousScenePath
+		{
+			get => EditorPrefs.GetString("SceneAutoLoader.PreviousScenePath", UnityEngine.SceneManagement.SceneManager.GetActiveScene().path);
+			set => EditorPrefs.SetString("SceneAutoLoader.PreviousScenePath", value);
+		}
+
+		// Static constructor binds a playmode-changed callback.
+		// [InitializeOnLoad] above makes sure this gets executed.
+		static SceneAutoLoader()
 		{
-			if (EditorSceneManager.GetActiveScene().name.Contains("InitTestScene"))
+			EditorApplication.playModeStateChanged += OnPlayModeChanged;
+		}
+
+		// Play mode change callback handles the scene load/reload.
+		private static void OnPlayModeChanged(PlayModeStateChange state)
+		{
+			if (state == PlayModeStateChange.ExitingEditMode)
 			{
-				EditorPrefs.SetString("prevEditorScene", "RRT CleanStation"); //Sets it to the Test statistician to load
+				OnEnteringPlayMode();
+			}
+
+			if (state == PlayModeStateChange.EnteredEditMode)
+			{
+				OnExitedPlayMode();
+			}
+		}
+
+		private static void OnEnteringPlayMode()
+		{
+			if (UnityEngine.SceneManagement.SceneManager.GetActiveScene().name.Contains("InitTestScene"))
+			{
+				PreviousSceneName = "RRT CleanStation"; //Sets it to the Test statistician to load
 				return; //tests are running do not interfere
 			}
 
-			if (EditorSceneManager.GetActiveScene().name == "Lobby" ||
-			    EditorSceneManager.GetActiveScene().name == "OnlineScene")
+			// User pressed play -- check for unsaved changes.
+			if (EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo() == false)
 			{
-				EditorPrefs.SetString("prevEditorScene", "");
-				PreviousScene = "";
+				// User cancelled the save operation -- cancel play as well.
+				EditorApplication.isPlaying = false;
 				return;
 			}
 
-			EditorPrefs.SetString("prevEditorScene", EditorSceneManager.GetActiveScene().name);
+			// Save current scene to return to it after play mode.
+			PreviousSceneName = UnityEngine.SceneManagement.SceneManager.GetActiveScene().name;
+			PreviousScenePath = UnityEngine.SceneManagement.SceneManager.GetActiveScene().path;
 
-			// User pressed play -- autoload online scene.
-			PreviousScene = EditorSceneManager.GetActiveScene().path;
-			if (EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo())
+			// if the scene is startup or lobby, load lobby scene
+			var initialScene = UnityEngine.SceneManagement.SceneManager.GetActiveScene().path;
+			if (initialScene.Contains("StartUp") || initialScene.Contains("Lobby"))
```

Comment:
what about online scene?

Reply:
If it's not lobby or startup, it'll load `MasterScene` which is the online scene

Reply:
ok cool